### PR TITLE
Add `rule_extension_apis_available`

### PR DESCRIPTION
**Pull request by hvadehra**

### CODE REVIEW

File: features.bzl [hunk 1]
```diff
@@ -97,6 +97,8 @@ _rules = struct(
     instrumented_files_info_has_metadata_files = ge("7.0.0-pre.20230710.5"),
     # Whether treeartifacts can have symlinks pointing outside of the tree artifact. (#21263)
     permits_treeartifact_uplevel_symlinks = ge("7.1.0"),
+    # Whether rule extension APIs are available by default
+    rule_extension_apis_available = ge("8.0.0"),
```
Comment (meteorcloudy) [inline features.bzl:101]: nit: would `8.0.0` include 8.0.0rc1? I forgot @fmeum

**Review by meteorcloudy — APPROVED** [review]

File: features.bzl (hunk 1 above)
Comment (fmeum) [reply features.bzl:101]: Yes, as far as I remember we made the check ignore RC suffixes so that we can test more realistically

Comment (fmeum) [reply features.bzl:101]: Let's wait for https://github.com/bazel-contrib/bazel_features/pull/81.

Comment (fmeum) [reply features.bzl:101]: Turns out we only did this for Bazel compatibility, not bazel_features. Which I guess makes sense.

Comment (Wyverald) [reply features.bzl:101]: we can just do `ge("8.0.0rc1")`, right?

Comment (fmeum) [reply features.bzl:101]: Yes, that's what we should do.

Comment (hvadehra) [reply features.bzl]: Updated to `8.0.0rc1`

**Pull request merged**